qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/snippets/QuteParentSnippetContext.java

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.services.snippets;
+
+import java.io.IOException;
+import java.util.Map;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import com.redhat.qute.parser.template.Node;
+import com.redhat.qute.parser.template.NodeKind;
+import com.redhat.qute.parser.template.Section;
+import com.redhat.qute.services.completions.CompletionRequest;
+
+/**
+ * Snippet context for Qute sections that have parent sections.
+ *
+ */
+
+public class QuteParentSnippetContext implements IQuteSnippetContext {

```

Comment:
`implements IQuteSnippetContext<CompletionRequest>`

Reply:
should IQuteSnippetContext be paramaterized, since ISnippetContext is paramatarized?